### ISSUE TITLE
alluxio.user.file.metadata.sync.interval does not work

### ISSUE THREAD

**Issue by theHarpers** [Alluxio/alluxio#8531]
**Alluxio Version:** <1.8.1>

**Describe the bug**
Alluxio master would always try to list files in UFS for to `alluxio fs ls /xxx` with **alluxio.user.file.metadata.sync.interval**>0.

**To Reproduce**
set **alluxio.user.file.metadata.sync.interval=60min** in alluxio-site.properties
execute `alluxio fs ls /xxx` several times, alluxio master would always sync with ufs.

**Expected behavior**
alluxio master would sync with ufs only for the first time.

**Additional context**
Seems **UfsSyncPathCache** is always empty. The `pathsToLoad` on `DefaultFileSystemMaster.java:3345` is always empty, thus **UfsSyncPathCache.notifySyncedPath()** is never executed.

**Comment by yuzhu** [comment]
@theHarpers
Hi, thanks for reporting your issues. 
A couple of clarifying questions, since I was not able to reproduce it the first time on my local machine. 

1. could you specify what UFS you are using? 
2. Did you restart Alluxio after changing `alluxio.user.file.metadata.sync.interval=60min` in the configuration file? Because this is a user configuration setting, Alluxio needs to restart to pick up this configuration setting. 
3. When you say "alluxio master would always sync with ufs", how did you observe that? did you create files in the UFS and see those files get listed under Alluxio?  

Thanks!

**Comment by theHarpers** [comment]
Hi, thanks for replying
1. I'm using s3 file system, and I observed this by setting log level of s3 request to DEBUG
2. I did restart several times after changing that setting
3. please see 1.

**Comment by theHarpers** [comment]
1.  Now I have a AlluxioMaster process running
```1988125791 35886     1   0 Wed05PM ttys000    0:56.32 /usr/bin/java -cp /projects/opt/alluxio-1.8.1/conf/::/projects/opt/alluxio-1.8.1/assembly/alluxio-server-1.8.1.jar -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=6607 -Dalluxio.home=/projects/opt/alluxio-1.8.1 -Dalluxio.conf.dir=/projects/opt/alluxio-1.8.1/conf -Dalluxio.logs.dir=/projects/opt/alluxio-1.8.1/logs -Dlog4j.configuration=file:/projects/opt/alluxio-1.8.1/conf/log4j.properties -Dorg.apache.jasper.compiler.disablejsr199=true -Djava.net.preferIPv4Stack=true -Dalluxio.logger.type=MASTER_LOGGER -Dalluxio.master.audit.logger.type=MASTER_AUDIT_LOGGER -Xmx8g alluxio.master.AlluxioMaster```

2. I set the `alluxio.user.file.metadata.sync.interval=5min` in `conf/alluxio-site.properties`, and added `log4j.logger.com.amazonaws.request=DEBUG` in `conf/log4j.properties`

3. I restarted the alluxio master:
```1988125791 45181     1   0  5:33PM ttys010    0:14.19 /usr/bin/java -cp /projects/opt/alluxio-1.8.1/conf/::/projects/opt/alluxio-1.8.1/assembly/alluxio-server-1.8.1.jar -Dalluxio.home=/projects/opt/alluxio-1.8.1 -Dalluxio.conf.dir=/projects/opt/alluxio-1.8.1/conf -Dalluxio.logs.dir=/projects/opt/alluxio-1.8.1/logs -Dlog4j.configuration=file:/projects/opt/alluxio-1.8.1/conf/log4j.properties -Dorg.apache.jasper.compiler.disablejsr199=true -Djava.net.preferIPv4Stack=true -Dalluxio.logger.type=MASTER_LOGGER -Dalluxio.master.audit.logger.type=MASTER_AUDIT_LOGGER -Xmx8g alluxio.master.AlluxioMaster
1988125791 45202 43236   0  5:35PM ttys014    0:00.00 grep alluxio```

4. execute

**Comment by theHarpers** [comment]
If you did not see this happening ,please restart the AlluxioMaster after the first `ls` command

**Comment by theHarpers** [comment]
And sorry about that I closed this

**Comment by yuzhu** [comment]
@theHarpers thanks for your detailed instruction. I was able to reproduce this in the 1.8 branch with your help. 

This issue has already been addressed in the master branch and in version 2.0 with #7960.  I will back port the fix to the 1.8 branch soon. 
Note this is an efficiency issue rather than a correctness one, we are accessing the UFS unnecessarily when we do not need to.

**Comment by apc999** [comment]
will close this issue since it has been addressed in 2.0-preview, 2.0-SNAPSHOT branch and cherry-picked to 1.8 branch by #8561 .
